crates/deltalake-core/src/operations/merge.rs

```diff
@@ -2569,17 +2570,43 @@ mod tests {
 
         assert!(pred.is_some());
 
-        let expected_pred = ["C", "X", "B"]
-            .into_iter()
-            .map(|id| {
-                lit(ScalarValue::Utf8(id.to_owned().into())).eq(col(Column {
-                    relation: Some(target_name.clone()),
-                    name: "id".to_owned(),
-                }))
-            })
-            .reduce(Expr::or)
-            .unwrap();
+        let split_pred = {
```

Comment:
You don't need to implement this yourself. Datafusion provides a utility for this: https://docs.rs/datafusion/latest/datafusion/logical_expr/utils/fn.split_binary_owned.html

Reply:
i'll use that for the first layer unpacking the ORs. but I think I still need to do manual mangling on unwrapping the EQ expressions, because Expr doesn't implement `Ord` so I need to extract the strings to sort them?

Reply:
Ah alright if that's the case then I think this is good!

Reply:
`split_binary_owned` is new to v34: https://docs.rs/datafusion/33.0.0/datafusion/?search=split_binary_owned

if #1983  makes it first then I can update it. :)

Reply:
We can merge it now just want to give some time before I press the merge button. If you want to wait until after the other PR let me know.